portal/models/qbd.py

```diff
@@ -103,5 +105,16 @@ def completed_date(self, user_id):
             raise ValueError(
                 f"Should never find multiple completed for {user_id} {self}")
         if not query.count():
+            # Check indefinite case, which doesn't generate timeline rows
+            if self._questionnaire_bank.classification == 'indefinite':
+                query = QuestionnaireResponse.query.filter(
+                    QuestionnaireResponse.subject_id == user_id).filter(
+                    QuestionnaireResponse.status == 'completed').join(
+                    QuestionnaireBank).filter(
+                    QuestionnaireResponse.questionnaire_bank_id == QuestionnaireBank.id).filter(
+                    QuestionnaireBank.classification == 'indefinite').with_entities(
+                    QuestionnaireResponse.document['authored'].label('authored'))
+                if query.count():
+                    return FHIR_datetime.parse(query.first()[0])
```

Comment:
I think using both `query.count()` and `query.first()` will cause 2 queries to be run vs a single one if only `first()` is used.
```suggestion
                first = query.first()
                if first:
                    return FHIR_datetime.parse(first[0])
```

Reply:
good point!